sambacc/_xattr.py

```diff
@@ -0,0 +1,78 @@
+#
+# sambacc: a samba container configuration tool
+# Copyright (C) 2022  John Mulligan
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>
+#
+"""xattr shim module
+
+This module exists to insulate sambacc from the platform xattr module.
+Currently it only support pyxattr. This module can be imported without
+pyxattr (xattr) present. The functions will import the required module
+and raise an ImportError if xattr is not available.
+
+This shim also provides a typed functions for xattr management. This
+could have been accomplished by writing a pyi file for xattr but since
+we need the runtime support we just add new functions.
+"""
+
+
+import pathlib
+import typing
+
+XAttrItem = typing.Union[
+    int,
```

Comment:
I stared at this `int` for few minutes, thinking t myself "how could you pass int to xattr.get?" until i realized this int is an open file-descriptor :)
Code looks good. My only comment is to add a comment next to this int field (some thing like `# an open fd`)

Reply:
:-)  will do

Reply:
done.